client/blocks/reader-full-post/index.jsx

```diff
@@ -487,7 +490,11 @@ export default connect(
 
 		const { site_ID: siteId, is_external: isExternal } = post;
 
-		const props = { post };
+		const props = {
+			post,
+			likeCount: getPostLikeCount( state, siteId, postId ),
+			liked: isLikedPost( state, siteId, postId ),
```

Comment:
I follow a few P2s in Reader and the posts' URLs are in format `/read/feeds/23251994/posts/1802580927`. In such a case, likes are broken.

`postId` prop is different from `prop.postId` and `isLikedPost` selector is using the wrong one. `likeCount` is always 0 and `liked` is always `false` for me.

(`likeCount` prop is not used anywhere in the component)

Displaying the `ReaderLikeButton` is OK, because it does its own `connect` and `QueryPostLikes`, but the keyboard shortcut to unlike a post (`l`) doesn't work. The component thinks I don't like the post yet and tries to like it.

Reply:
Yeah, same issue as above. We need to pull the site ID and post ID off the post object to be safe, unless we have a explicit blog id

Reply:
fixed in https://github.com/Automattic/wp-calypso/pull/23570/commits/1e69954fcc1ce030622e020b2012ecf636dcde82